test/Test.re

```diff
@@ -1,1061 +1,716 @@
+                subTreeChanged: `NoChange,
+                oldInstance:
+                  <ChangeCounter id=1 label="updatedText" counter=11 />,
+                newInstance:
+                  <ChangeCounter id=1 label="updatedText" counter=2011 />
+              })
+            ]
+          ),
+          actual2f
+        );
+        let (rendered2f_mem, _) =
+          ReasonReact.RenderedElement.flushPendingUpdates(rendered2f);
+        check(
+          Alcotest.bool,
+          "it is memoized",
+          rendered2f_mem === rendered2f,
+          true
+        );
+        let (rendered2f_mem, _) =
+          ReasonReact.RenderedElement.flushPendingUpdates(rendered2f_mem);
+        check(
+          Alcotest.bool,
+          "it is memoized",
+          rendered2f_mem === rendered2f,
+          true
+        );
+        let (rendered3, _) as actual3 =
+          TestRenderer.update(
+            rendered2f_mem,
+            <ButtonWrapperWrapper wrappedText="updatedText" />
+          );
+        assertUpdate(
+          ~label="Updating components: ChangeCounter to ButtonWrapperWrapper",
+          TestComponents.(
+            [
+              <ButtonWrapperWrapper id=2 nestedText="wrappedText:updatedText" />
+            ],
+            Some(
+              TestRenderer.{
+                subtreeChange: `Nested,
+                updateLog:
+                  ref([
+                    ChangeComponent({
+                      oldSubtree: [],
+                      newSubtree: [
+                        <Div id=3>
+                          <Text id=4 title="buttonWrapperWrapperState" />
+                          <Text id=5 title="wrappedText:updatedText" />
+                          <ButtonWrapper id=6 />
+                        </Div>
+                      ],
+                      oldInstance:
+                        <ChangeCounter id=1 label="updatedText" counter=2011 />,
+                      newInstance:
+                        <ButtonWrapperWrapper
+                          id=2
+                          nestedText="wrappedText:updatedText"
+                        />
+                    })
+                  ])
+              }
+            )
+          ),
+          actual3
+        );
+        let (rendered4, _) as actual4 =
+          TestRenderer.update(
+            rendered3,
+            <ButtonWrapperWrapper wrappedText="updatedTextmodified" />
+          );
+        assertUpdate(
+          ~label="Updating text in the button wrapper",
+          TestComponents.(
+            [
+              <ButtonWrapperWrapper
+                id=2
+                nestedText="wrappedText:updatedTextmodified"
+              />
+            ],
+            Some(
+              TestRenderer.{
+                subtreeChange: `Nested,
+                updateLog:
+                  ref([
+                    UpdateInstance({
+                      stateChanged: true,
+                      subTreeChanged: `NoChange,
```

Comment:
This doesn't make sense - the subtree didn't change but the content has changed. It'll probably require something like `shouldUpdateContent` on the `nativeElement`... (Although it'd be awesome to avoid that.)